ui/src/app/utils/leo-runtime-initializer.tsx

```diff
@@ -196,16 +194,16 @@ export class LeoRuntimeInitializer {
       throw new ExceededActionCountError(
         `Reached max runtime create count (${this.maxCreateCount})`, this.currentRuntime);
     }
+
     let runtime: Runtime;
     if (serverConfigStore.getValue().enableCustomRuntimes && this.targetRuntime) {
       runtime = this.targetRuntime;
+    } else if (this.currentRuntime) {
```

Comment:
your comment about factoring out the business logic made me realize that I could probably just remove that check here.

In most cases the status will be Deleted at this point but even in the case where it isn't, I feel like it probably makes sense to create the currentRuntime (w/ the preset override) instead of defaulting to the General Analysis template.